scripts/cc-cli/src/test/integration.test.ts

```diff
@@ -0,0 +1,404 @@
+      .isTrue;
+    expect(nodeHasKeys).toBe(true);
+
+    // Signaling intention to validate should make the validator (stash) appear as waiting
+    const signalResult = execa.commandSync(
+      `creditcoin-cli validate --commission 1 --url ws://localhost:9945`,
+      {
+        env: {
+          CC_CONTROLLER_SEED: controllerSeed,
+        },
+      }
+    );
+
+    const stashStatusAfterValidating = await getStatus(stashAddress, bobApi);
+    expect(stashStatusAfterValidating.waiting).toBe(true);
+
+    // After increasing the validator count, (forcing an era- currently not) and waiting for the next era,
+    // the validator should become elected & active.
+    const increaseValidatorCountTx = aliceApi.tx.staking.setValidatorCount(2);
+    const increaseValidatorCountSudoTx = aliceApi.tx.sudo.sudo(
+      increaseValidatorCountTx
+    );
+    await signSendAndWatch(
+      increaseValidatorCountSudoTx,
+      aliceApi,
+      initKeyringPair("//Alice")
+    );
+    const validatorCount = await (
+      await aliceApi.query.staking.validatorCount()
+    ).toNumber();
+    expect(validatorCount).toBe(2);
+    await waitEras(2, aliceApi);
+    const stashStatusAfterEra = await getStatus(stashAddress, bobApi);
+    expect(stashStatusAfterEra.active).toBe(true);
+
+    // After waiting for another era, the validator should have accumulated era rewards to distribute
+    const startingEra = (
+      await aliceApi.derive.session.info()
+    ).activeEra.toNumber();
+    console.log("Starting era: ", startingEra);
+    await waitEras(1, aliceApi);
+
+    // const eraRewards = await aliceApi.query.staking.erasRewardPoints(
+    //   startingEra
+    // );
+    // expect(eraRewards.individual.entries()).toBeGreaterThan(0);
+
+    // After distributing rewards, the validator staked balance should increase
+    // (because it was set to staked)
+    const balanceBeforeRewards = await getBalance(stashAddress, aliceApi);
+    console.log(balanceBeforeRewards.bonded.toString());
+
+    const distributeCommand = execa.commandSync(
+      `creditcoin-cli distribute-rewards --url ws://localhost:9945 --validator-id ${stashAddress} --era ${startingEra}`,
+      {
+        env: {
+          CC_SEED: stashSeed,
+        },
+      }
+    );
+
+    // wait 5 seconds for nodes to sync
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+    const balanceAfterRewards = await getBalance(stashAddress, aliceApi);
+    console.log(balanceAfterRewards.bonded.toString());
+    const balanceIncreased = balanceAfterRewards.bonded.gt(
+      balanceBeforeRewards.bonded
+    );
+    expect(balanceIncreased).toBe(true);
+
+    // After executing the chill commmand, the validator should no longer be active nor waiting
+    const chillCommand = execa.commandSync(
+      `creditcoin-cli chill --url ws://localhost:9945`,
+      {
+        env: {
+          CC_CONTROLLER_SEED: controllerSeed,
+        },
+      }
+    );
+    // wait 5 seconds for nodes to sync
```

Comment:
I think split this file into multiple files describing the different scenarios here. Otherwise it becomes too big.